GO_SOPT_Seminar_Assignment/SecondSeminar/FirstViewController_2nd_Assignment.swift

```diff
@@ -0,0 +1,380 @@
+            $0.leading.equalTo(whereIDButton.snp.trailing).offset(80)
+        }
+        
+        barLabel.snp.makeConstraints {
+            $0.centerY.equalTo(whereIDButton.snp.centerY)
+            $0.leading.equalTo(whereIDButton.snp.trailing).offset(40)
+        }
+        
+        yetLabel.snp.makeConstraints {
+            $0.top.equalToSuperview().inset(423)
+            $0.leading.equalToSuperview().offset(51)
+        }
+        
+        makeIDButton.snp.makeConstraints {
+            $0.top.equalToSuperview().inset(423)
+            $0.centerY.equalTo(yetLabel.snp.centerY)
+            $0.leading.equalTo(yetLabel.snp.trailing).offset(42)
+        }
+        
+        backButton.snp.makeConstraints {
+            $0.top.equalToSuperview().inset(65)
+            $0.leading.equalToSuperview().offset(19)
+            $0.height.equalTo(15)
+            $0.width.equalTo(8)
+        }
+    }
+    
+    // MARK: - 로그인 가능 조건
+    
+    func canLogin() {
+        let tOrf = idTextField.hasText && passwordTextField.hasText
+        updateLoginButton(willActive: tOrf)
+    }
+    
+    // MARK: - 로그인버튼 활성화 or 비활성화
+    
+    func updateLoginButton(willActive: Bool) {
+        if willActive {
+            setLoginButton(color: .tvingRed, width: 0, tOrf: true)
+            loginButton.setTitleColor(.white, for: .normal)
+        } else {
+            setLoginButton(color: .black, width: 1, tOrf: false)
+            loginButton.setTitleColor(.tvingGray3, for: .normal)
+        }
+    }
+    
+    // MARK: - 로그인버튼 활성화 or 비활성화일 때 특성 주는 함수
+    
+    func setLoginButton(color: UIColor, width: CGFloat, tOrf: Bool) {
+        loginButton.backgroundColor = color
+        loginButton.makeBorder(width: width, color: .tvingGray3)
+        loginButton.isEnabled = tOrf
+    }
+    
+    // MARK: - eyeButton & deleteButton
+    
+    func whenIdInput() {
+        eyeButton.isHidden = true
+        eyeButton.isEnabled = false
+        deletePasswordButton.isHidden = true
+        deletePasswordButton.isEnabled = false
+        
+        deleteIDButton.isHidden = false
+        deleteIDButton.isEnabled = true
+        deleteIDButton.addTarget(self, action: #selector(deleteIdButtonTapped), for: .touchUpInside)
+    }
+    
+    func whenPasswordInput() {
+        deleteIDButton.isHidden = true
+        deleteIDButton.isEnabled = false
+
+        eyeButton.isHidden = false
+        eyeButton.isEnabled = true
+        eyeButton.addTarget(self, action: #selector(eyeButtonTapped), for: .touchUpInside)
+        
+        deletePasswordButton.isHidden = false
+        deletePasswordButton.isEnabled = true
+        deletePasswordButton.addTarget(self, action: #selector(deletePasswordButtonTapped), for: .touchUpInside)
+    }
+    
+    @objc
+    func eyeButtonTapped() {
+        if passwordTextField.isSecureTextEntry {
+            passwordTextField.isSecureTextEntry = false
+        } else {
+            passwordTextField.isSecureTextEntry = true
+        }
```

Comment:
eyeButton을 눌렀을때 true->false, false->true가 되는 상황에서는 이렇게 한줄로 작성할 수도 있답니다 !! 

```suggestion
         passwordTextField.isSecureTextEntry.toggle()
```

Reply:
와우...대벅이네요 감사해요🥲